client/platform/desktop/backend/native/common.ts

```diff
@@ -430,36 +441,81 @@ async function importMedia(settings: Settings, path: string): Promise<JsonMeta>
 
   const contents = await fs.readdir(jsonMeta.originalBasePath);
 
+  let mediaConvertList: string[] = [];
   /* Extract and validate media from import path */
   if (jsonMeta.type === 'video') {
     jsonMeta.originalVideoFile = npath.basename(path);
     const mimetype = mime.lookup(path);
     if (mimetype) {
       if (websafeImageTypes.includes(mimetype) || otherImageTypes.includes(mimetype)) {
         throw new Error('User chose image file for video import option');
-      } else if (websafeVideoTypes.includes(mimetype)) {
-        /* TODO: Kick off video inspection and maybe transcode */
+      } else if (websafeVideoTypes.includes(mimetype) || otherVideoTypes.includes(mimetype)) {
+        if (!checkMedia(settings, path) || otherVideoTypes.includes(mimetype)) {
+          mediaConvertList.push(path);
+        }
       } else {
         throw new Error(`unsupported MIME type for video ${mimetype}`);
       }
     } else {
       throw new Error(`could not determine video MIME type for ${path}`);
     }
   } else if (datasetType === 'image-sequence') {
-    jsonMeta.originalImageFiles = contents.filter((filename) => {
+    const tempConvertList: string[] = [];
+    let convertAny = false; //If we have to convert one image we convert all for organization
+    contents.forEach((filename) => {
       const abspath = npath.join(jsonMeta.originalBasePath, filename);
       const mimetype = mime.lookup(abspath);
-      /* TODO: support transcoding of non-web-safe image types */
-      return !!(mimetype && websafeImageTypes.includes(mimetype));
+      if (mimetype && (websafeImageTypes.includes(mimetype)
+      || otherImageTypes.includes(mimetype))) {
+        jsonMeta.originalImageFiles.push(filename);
+        tempConvertList.push(abspath);
+        if (otherImageTypes.includes(mimetype)) {
+          convertAny = true;
+        }
+      }
     });
     if (jsonMeta.originalImageFiles.length === 0) {
       throw new Error(`no images found in ${path}`);
     }
+    if (convertAny) {
+      mediaConvertList = tempConvertList;
+    }
   } else {
     throw new Error('only video and image-sequence types are supported');
   }
 
   const projectDirAbsPath = await _initializeProjectDir(settings, jsonMeta);
+
+  //Now we will kick off any conversions that are necessary
+  let jobBase = null;
+  if (mediaConvertList.length) {
+    const srcDstList: [string, string][] = [];
+    const extension = datasetType === 'video' ? '.mp4' : '.png';
+    mediaConvertList.forEach((item) => {
+      const destLoc = item.replace(jsonMeta.originalBasePath, projectDirAbsPath);
+      const destExt = destLoc.replace(npath.extname(item), extension);
```

Comment:
This is a confusing name.  `destExt` isn't an ext, it's the entire absolute path.  Could you make this a more expressive variable name?